src/shares/PaymentHub.sol

```diff
@@ -46,14 +49,71 @@ contract PaymentHub {
     
     IQuoter constant uniswapQuoter = IQuoter(0xb27308f9F90D607463bb33eA1BeBb41C27CE5AB6);
     ISwapRouter constant uniswapRouter = ISwapRouter(0xE592427A0AEce92De3Edee1F18E0157C05861564);
+    AggregatorV3Interface internal priceFeedCHFUSD;
+    AggregatorV3Interface internal priceFeedETHUSD;
 
-    constructor(address currency_) {
-        currency = currency_;
+    constructor(address _currency, address _aggregatorCHFUSD, address _aggregatorETHUSD) {
+        currency = _currency;
         weth = uniswapQuoter.WETH9();
+        priceFeedCHFUSD = AggregatorV3Interface(_aggregatorCHFUSD);
+        priceFeedETHUSD = AggregatorV3Interface(_aggregatorETHUSD);
+
     }
 
     function getPriceInEther(uint256 amountOfXCHF) external returns (uint256) {
-        return uniswapQuoter.quoteExactOutputSingle(weth, currency, 3000, amountOfXCHF, 0);
+        return getPriceInEther(amountOfXCHF, address(0));
+    }
+
+    /**
+     * Get price in Ether depding on brokerbot setting.
+     * If keep ETH is set price is from oracle.
+     */
+    function getPriceInEther(uint256 amountOfXCHF, address brokerBot) public returns (uint256) {
+        if ((brokerBot != address(0)) && hasSettingKeepEther(brokerBot)) {
```

Comment:
Why do we check against address(0) here, and not on line 177? 
I'm just curious against what scenario this protects vs. what minimal cost it adds.